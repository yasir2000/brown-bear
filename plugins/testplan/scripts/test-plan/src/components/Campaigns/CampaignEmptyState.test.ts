/*
 * Copyright (c) Enalean, 2020 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import type { Wrapper } from "@vue/test-utils";
import { shallowMount } from "@vue/test-utils";
import CampaignEmptyState from "./CampaignEmptyState.vue";
import { createTestPlanLocalVue } from "../../helpers/local-vue-for-test";
import { createStoreMock } from "../../../../../../../src/scripts/vue-components/store-wrapper-jest";
import type { RootState } from "../../store/type";

describe("CampaignEmptyState", () => {
    async function createWrapper(
        user_can_create_campaign: boolean,
        show_create_modal = jest.fn()
    ): Promise<Wrapper<CampaignEmptyState>> {
        return shallowMount(CampaignEmptyState, {
            localVue: await createTestPlanLocalVue(),
            mocks: {
                $store: createStoreMock({
                    state: {
                        user_can_create_campaign,
                    } as RootState,
                }),
            },
            propsData: {
                showCreateModal: show_create_modal,
            },
        });
    }

    it("Displays empty state with new campaign creation button", async () => {
        const wrapper = await createWrapper(true);

        expect(wrapper.element).toMatchSnapshot();
        expect(wrapper.find("[data-test=new-campaign]").exists()).toBe(true);
    });

    it("Does not show the new campaign creation button in the empty state when the user cannot create new ones", async () => {
        const wrapper = await createWrapper(false);

        expect(wrapper.find("[data-test=new-campaign]").exists()).toBe(false);
    });

    it(`On click on the button, it calls showCreateModal`, async () => {
        const show_create_modal = jest.fn();

        const wrapper = await createWrapper(true, show_create_modal);

        await wrapper.find("[data-test=new-campaign]").trigger("click");

        expect(show_create_modal).toHaveBeenCalled();
    });
});
