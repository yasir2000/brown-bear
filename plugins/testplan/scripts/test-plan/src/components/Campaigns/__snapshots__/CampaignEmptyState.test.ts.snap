// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignEmptyState Displays empty state with new campaign creation button 1`] = `
<section
  class="empty-state-pane"
>
  <div
    class="empty-state-illustration"
  >
    <campaign-empty-state-svg-stub />
  </div>
   
  <p
    class="empty-state-text"
  >
    <translate-stub
      tag="span"
    >
      There is no test campaign yet
    </translate-stub>
  </p>
   
  <button
    class="tlp-button-primary tlp-button-outline empty-state-action"
    data-test="new-campaign"
    type="button"
  >
    <i
      class="fa fa-plus tlp-button-icon"
    />
     
    <translate-stub
      tag="span"
    >
      Create new campaign
    </translate-stub>
  </button>
</section>
`;
